lib/ddtrace/profiling/transport/http.rb

```diff
@@ -16,15 +16,13 @@ module Profiling
     module Transport
       # Namespace for HTTP transport components
       module HTTP
-        module_function
```

Comment:
why doesn't sorbet like `module_function` 😢

Reply:
I think it's because of the "call from outside"+"call from inside" behavior:

```ruby
[1] pry(main)> module Foo
[1] pry(main)*   module_function
[1] pry(main)*   def foo
[1] pry(main)*     :foo
[1] pry(main)*   end
[1] pry(main)* end
=> :foo
[2] pry(main)> class MyClass
[2] pry(main)*   include Foo
[2] pry(main)*   def call_foo
[2] pry(main)*     foo
[2] pry(main)*   end
[2] pry(main)* end
=> :call_foo
[3] pry(main)> MyClass.new.call_foo
=> :foo
```

but yeah _waves hands_ this one was mostly "sure, sorbet, sure".